libs/ui/dataviz/src/lib/table/table.component.ts

```diff
@@ -33,38 +39,67 @@ export interface TableItemModel {
   imports: [
     MatTableModule,
     MatSortModule,
-    TableVirtualScrollModule,
+    MatPaginatorModule,
     ScrollingModule,
     NgForOf,
     TranslateModule,
   ],
+  providers: [{ provide: MatPaginatorIntl, useClass: CustomMatPaginatorIntl }],
   selector: 'gn-ui-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TableComponent implements AfterViewInit {
-  @Input() set data(value: TableItemModel[]) {
-    this.dataSource = new TableVirtualScrollDataSource(value)
-    this.dataSource.sort = this.sort
-    this.properties =
-      Array.isArray(value) && value.length ? Object.keys(value[0]) : []
-    this.count = value.length
+export class TableComponent implements OnInit, AfterViewInit, OnChanges {
```

Comment:
What do you think of renaming this component `DataTableComponent` ?

Reply:
Hum, and then I rename the `TableScrollComponent` back to `TableComponent` which it actually was in the beginning? Would that make sense to you, a `TableComponent` that accepts datasets and a `DataTableComponent` that accepts data readers?

Reply:
I think we can just get rid of the table scroll component, unless it has some use still?

Reply:
I thought to keep it temporarily for the `geo-table-view.component` which has various issues to work with the new `TableComponent`.